src/main/java/com/genesys/roberta/tokenizer/RobertaTokenizer.java

```diff
@@ -0,0 +1,108 @@
+package com.genesys.roberta.tokenizer;
+
+import lombok.NonNull;
+import lombok.val;
+
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.LongStream;
+
+import static java.util.stream.LongStream.concat;
+import static java.util.stream.LongStream.of;
+
+/**
+ * Tokenizer used for the RoBERTa model.
+ * Encode sentences to integer tokens.
+ *
+ * This tokenizer is implemented according to the following:
+ * - https://huggingface.co/docs/transformers/model_doc/roberta#transformers.RobertaTokenizer
+ * - https://github.com/hyunwoongko/gpt2-tokenizer-java/blob/master/src/main/java/ai/tunib/tokenizer/GPT2Tokenizer.java
+ * - https://github.com/huggingface/tflite-android-transformers/blob/master/gpt2/src/main/java/co/huggingface/android_transformers/\
+ *   gpt2/tokenization/GPT2Tokenizer.kt
+ */
+public class RobertaTokenizer implements Tokenizer {
+
+    public static final long PAD_TOKEN = 1; // Tokenized sentences shorter than the max allowed length will be padded with the PAD_TOKEN
+
+    public static final long CLS_TOKEN = 0; // Classification token. Also BOS (beginning of sequence) token
+
+    public static final long SEP_TOKEN = 2; // Separator token. Also EOS (end of sequence) token
+
+    public static final long UNK_TOKEN = 3; // Unknown Token.
+
+    //splits a given sentence by space in to words or sub-words
+    private static final Pattern PATTERN = Pattern
+            .compile("'s|'t|'re|'ve|'m|'ll|'d| ?\\p{L}+| ?\\p{N}+| ?[^\\s\\p{L}\\p{N}]+|\\s+(?!\\S)|\\s+");
+
+    private final RobertaTokenizerResourcesFactory robertaTokenizerFactory;
+    private final AtomicReference<RobertaTokenizerResources> robertaResourcesCache = new AtomicReference<>();
```

Comment:
Just for the record - we'll do a more thorough review later - does the fact that `AtomicReference` is used here, means that clients of this class can be sure they always have one `RobertaTokenizerResources` instance at hand, and thus the client doesn't need to also use such a mechanism?